packages/flutter/lib/src/widgets/tap_region.dart

```diff
@@ -403,6 +404,8 @@ class TapRegion extends SingleChildRenderObjectWidget {
 
   @override
   RenderObject createRenderObject(BuildContext context) {
+    ModalRoute.of(context);
```

Comment:
why didn't we disable on tap outside in the create method?

Reply:
also somewhere in the doc should mention this now will be aware of the route

Reply:
> why didn't we disable on tap outside in the create method?

I believe the assumption is that new widgets are always inflated in the active/current route.
(But now that you mention it, maybe there are edge cases we should take into account.)

Reply:
You may have cases where you have multiple initial routes. or multiple route add/push.

while they will not be laid out, but it will still be built

Reply:
Also I meant the API doc of TapRegion class should be updated to mention the new behavior

Reply:
> You may have cases where you have multiple initial routes. or multiple route add/push.
> 
> while they will not be laid out, but it will still be built
Hmm, I see.